pkg/controllermanager/controller/shoot/shoot_reference_control.go

```diff
@@ -68,32 +69,41 @@ func (c *Controller) shootReferenceUpdate(oldObj, newObj interface{}) {
 	}
 }
 
-func refChange(oldShoot, newShoot *gardencorev1beta1.Shoot) bool {
```

Comment:
Can you keep `refChange` and call all the relevant functions here? This will prevent a more and more expanding OR chain [here](https://github.com/gardener/gardener/pull/3071/files#diff-a03ba9a488c2d4f8e80461ff9f802de0fb38c91fc4c695c21636b4b5767426ffR61).